### PR TITLE
fix(45927): fix missing `path` for Module in v12

### LINKED ISSUE #45927: webpack-env missing path
I'm using the latest typescript version and due to the addition from **path** to node in 12.12.48 I get this error in webpack-env:

```
node_modules/@types/webpack-env/index.d.ts:349:15 - error TS2430: Interface 'NodeJS.Module' incorrectly extends interface '__WebpackModuleApi.Module'.
  Types of property 'parent' are incompatible.
    Type 'Module | null' is not assignable to type 'NodeModule | null'.
      Property 'path' is missing in type 'Module' but required in type 'NodeModule'.

349     interface Module extends __WebpackModuleApi.Module {}
                  ~~~~~~

  node_modules/@types/node/globals.d.ts:236:5
    236     path: string;
            ~~~~
    'path' is declared here.
```

@use-strict @rhonsby

### PR DESCRIPTION
**Pull request by peterblazejewicz**
This is an attempt to fix a problem reported in #45927

/cc @tk-1io

Thanks!

Fixes #45927

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)

### CODE REVIEW

**Review by ExE-Boss — APPROVED** [review]

File: types/node/v12/node-tests.ts [hunk 1]
```diff
@@ -764,6 +764,7 @@ import moduleModule = require('module');
     const m2: Module = new Module.Module("moduleId");
     const b: string[] = Module.builtinModules;
     let paths: string[] = module.paths;
+    const path = module.path;
```
Comment (ExE-Boss) [inline types/node/v12/node-tests.ts:767]:
```suggestion
    const path: string = module.path;
```

**Comment by tk-1io** [comment]
@peterblazejewicz Thanks!
I wonder if this fix should also be applied to v11? `path` has been added to `NodeModule` interface for v11 also: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/45858/files#diff-92bffea8910ef9d75a1c80057ca68b3cR232
But it seems there is also the `Module` class here and it is missing `path`, like in v12.

**Comment by tk-1io** [comment]
@peterblazejewicz Another idea... ;-) How about adding `implements NodeModule` to the `Module` class?
```
class Module implements NodeModule {
```
This should avoid further issues of this kind, as one cannot change the `NodeModule` interface and forget about the `Module` class?

**Comment by Flarna** [comment]
> How about adding `implements NodeModule` to the `Module` class?

As far as I remember we had this in the past but it was a pain as there are other modules augmenting `interface NodeModule` with more mandatory properties resulting in failed builds as `class Module` defined here has not included them.

But maybe this has been fixed meanwhile.

**Review by Flarna — APPROVED** [review]

**Review by a-tarasyuk — APPROVED** [review]

**Review by r3nya — APPROVED** [review]

**Comment by ExE-Boss** [comment]
You can do:
```ts
interface Module extends NodeJS.NodeModule {}
class Module {
	//
}
```

This is done by the `EventEmitter` class: <https://github.com/DefinitelyTyped/DefinitelyTyped/blob/a92d898478072a57dabc21308f7d49698b5c1be8/types/node/events.d.ts#L38-L41>

**Review by trivikr — APPROVED** [review]

**Comment by davidje13** [comment]
Same problem seems to have reappeared with `@types/node@14.10.1`;

```
node_modules/@types/node/ts3.1/globals.d.ts:573:15 - error TS2430: Interface 'NodeJS.Module' incorrectly extends interface '__WebpackModuleApi.Module'.
  Types of property 'parent' are incompatible.
    Type 'Module | null | undefined' is not assignable to type 'NodeModule | null'.
      Type 'undefined' is not assignable to type 'NodeModule | null'.

573     interface Module {
                  ~~~~~~


Found 1 error.
```

From a quick scan of this thread it seems there's some kind of issue with adding an explicit dependency here that would catch this problem before publishing, but would it be possible to add a unit test? It seems this same bit is breaking quite frequently (this looks like at least the third time it has been broken).

**Comment by ExE-Boss** [comment]
@davidje13 That’s correct because [`Module.parent`](https://nodejs.org/api/modules.html#modules_module_parent) returns `undefined` if the parent is not a **CommonJS** module, and is deprecated as a result.

**Comment by davidje13** [comment]
@ExE-Boss does that make this a bug in the webpack definitions then?

**Pull request merged**